src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,238 @@
+
+        self._current_velocity_sub = rospy.Subscriber(
+            '/odometry/filtered', Odometry,
+            self._current_velocity_callback)
+
+        try:
+            self._TRANSFORM_TIMEOUT = rospy.get_param('~transform_timeout')
+            self._MAX_HORIZ_SPEED = rospy.get_param('~max_translation_speed')
+            self._MAX_START_TASK_DIST = rospy.get_param('~roomba_max_start_task_dist')
+            self._MAX_END_TASK_DIST = rospy.get_param('~roomba_max_end_task_dist')
+            self._MAX_Z_VELOCITY = rospy.get_param('~max_z_velocity')
+            self._K_X = rospy.get_param('~k_term_tracking_x')
+            self._K_Y = rospy.get_param('~k_term_tracking_y')
+            GO_TO_HEIGHT = rospy.get_param('~track_roomba_height')
+            _roomba_diameter = rospy.get_param('~roomba_diameter')
+            _drone_width = rospy.get_param('~drone_landing_gear_width')
+        except KeyError as e:
+            rospy.logerr('Could not lookup a parameter for go_to roomba task')
+            raise
+
+        self._z_holder = HeightHolder(GO_TO_HEIGHT)
+        self._height_checker = HeightSettingsChecker()
+        self._limiter = AccelerationLimiter()
+
+        self._overshoot = 0.0
+
+        self._state = GoToObjectTaskState.init
+
+    def _receive_roomba_status(self, data):
+        with self._lock:
+            self._roomba_array = data
+
+    def _current_velocity_callback(self, data):
+        with self._lock:
+            self._drone_odometry = data
+
+    def get_desired_command(self):
+        with self._lock:
+            if self._canceled:
+                return (TaskCanceled(),)
+
+            elif (self._state == GoToObjectTaskState.init):
+                if self._roomba_array is None or self._drone_odometry is None:
+                    self._state = GoToObjectTaskState.waiting
+                else:
+                    self._state = GoToObjectTaskState.going
+                return (TaskRunning(), NopCommand())
+
+            elif (self._state == GoToObjectTaskState.waiting):
+                if self._roomba_array is None or self._drone_odometry is None:
+                    self._state = GoToObjectTaskState.waiting
+                else:
+                    self._state = GoToObjectTaskState.going
+                return (TaskRunning(), NopCommand())
+
+            elif self._state == GoToObjectTaskState.going:
+
+                if not (self._height_checker.above_min_maneuver_height(
+                            self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Drone is too low'),)
+                elif not (self._z_holder.check_z_error(
+                    self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Z error is too high'),)
+                elif not self._check_roomba_in_sight():
+                    return (TaskAborted(msg='The provided roomba is not in sight of quad'),)
+
+                try:
```

Comment:
@amiller27 the roomba odometry will give us the roomba's map position, no? If so, `self._roomba_array` should have that position. 

So I'm thinking is we use the roomba's global positions, but avoid using any estimates on the roomba's velocity since we may be too far to know which direction its moving. So here lets not find the vector from us to the roomba, but just use the position of the roomba in the map frame.

Reply:
Yes, the roomba odometry array is currently set up to include positions in the `map` frame, and I think that's probably how it's going to stay.